common/src/main/java/com/microsoft/identity/common/exception/CommonCoreBaseException.java

```diff
@@ -0,0 +1,149 @@
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.exception;
+
+import com.microsoft.identity.common.adal.internal.net.HttpWebResponse;
+import com.microsoft.identity.common.adal.internal.util.StringExtensions;
+import com.microsoft.identity.common.adal.internal.util.HashMapExtensions;
+
+import org.json.JSONException;
+
+import java.util.HashMap;
+import java.util.List;
+
+public class CommonCoreBaseException extends Exception {
+    private String mErrorCode;
+
+    /**
+     * Default constructor.
+     */
+    CommonCoreBaseException() {
+    }
+
+    /**
+     * Initiates the detailed error code.
+     *
+     * @param errorCode The error code contained in the exception.
+     */
+    public CommonCoreBaseException(final String errorCode) {
+        mErrorCode = errorCode;
+    }
+
+    /**
+     * Initiates the {@link CommonCoreBaseException} with error code and error message.
+     *
+     * @param errorCode    The error code contained in the exception.
+     * @param errorMessage The error message contained in the exception.
+     */
+    public CommonCoreBaseException(final String errorCode, final String errorMessage) {
+        super(errorMessage);
+        mErrorCode = errorCode;
+    }
+
+    /**
+     * Initiates the {@link CommonCoreBaseException} with error code, error message and throwable.
+     *
+     * @param errorCode    The error code contained in the exception.
+     * @param errorMessage The error message contained in the exception.
+     * @param throwable    The {@link Throwable} contains the cause for the exception.
+     */
+    public CommonCoreBaseException(final String errorCode, final String errorMessage,
+                  final Throwable throwable) {
+        super(errorMessage, throwable);
+        mErrorCode = errorCode;
+    }
+
+    /**
+     * @return The error code for the exception, could be null. {@link CommonCoreBaseException} is the top level base exception, for the
+     * constants value of all the error code.
+     */
+    public String getErrorCode() {
+        return mErrorCode;
+    }
+
+    /**
+     * {@inheritDoc}
+     * Return the detailed description explaining why the exception is returned back.
+     */
+    @Override
+    public String getMessage() {
+        if (!StringExtensions.isNullOrBlank(super.getMessage())) {
+            return super.getMessage();
+        }
+
+        return null;
+    }
+
+    protected int mHttpStatusCode;
+
+    protected HashMap<String, String> mHttpResponseBody = null;
```

Comment:
This field, `mHttpStatusCode` and `mHttpResponseHeaders` seem like they would be more suited on the `CommonCoreServiceException` subclass -- why does the base class contain these fields? It doesn't seem right to me that a client-side exception would have these fields

Reply:
+1

Reply:
you guys are right, it was just an attempt at avoiding exposing the 2 child classes to ADAL/MSAL. What do you guys think about the tradeoff?